blocks/api/serializer.js

```diff
@@ -115,7 +115,12 @@ export function serializeBlock( block ) {
 	const blockName = block.name;
 	const blockType = getBlockType( blockName );
 	const saveContent = getSaveContent( blockType, block.attributes );
-	const saveAttributes = getCommentAttributes( block.attributes, parseBlockAttributes( saveContent, blockType.attributes ) );
+	let saveAttributes = getCommentAttributes( block.attributes, parseBlockAttributes( saveContent, blockType.attributes ) );
+
+	// Remove attributes that are the same as the defaults.
+	if ( blockType.defaultAttributes ) {
+		saveAttributes = pickBy( saveAttributes, ( value, key ) => ! isEqual( value, blockType.defaultAttributes[ key ] ) );
```

Comment:
>In other words, `value !== blockType.defaultAttributes[ key ]` will always be `true` for objects.

Well, not necessarily; not if the attribute value is never updated. But it's a fair point that if they _are_ updated, and their nested structure matches the default value, it should probably be omitted all the same. I'm fine with `isEqual`.